src/tests/test_ucc_verifier.cpp

```diff
@@ -0,0 +1,153 @@
+
+    algos::StdParamsMap GetUCCVerifierParamsMap() const {
+        return ucc_verifier_params_map_;
+    }
+
+    algos::StdParamsMap GetHyUCCParamsMap() const {
+        return hyucc_params_map_;
+    }
+};
+
+class TestUCCVerifierWithHyUCC : public ::testing::TestWithParam<UCCVerifierWithHyUCCParams> {};
+
+TEST_P(TestUCCVerifierSimple, DefaultTest) {
+    UCCVerifierSimpleParams const& p(GetParam());
+    auto verifier = algos::CreateAndLoadAlgorithm<algos::UCCVerifier>(p.GetParamsMap());
+    verifier->Execute();
+    EXPECT_EQ(verifier->UCCHolds(), p.GetExpectedNumClustersViolatingUCC() == 0);
+    EXPECT_EQ(verifier->GetNumRowsViolatingUCC(), p.GetExpectedNumRowsViolatingUCC());
+    EXPECT_EQ(verifier->GetNumClustersViolatingUCC(), p.GetExpectedNumClustersViolatingUCC());
+    EXPECT_EQ(verifier->GetClustersViolatingUCC(), p.GetExpectedClustersViolatingUCC());
+}
+
+INSTANTIATE_TEST_SUITE_P(
+        UCCVerifierSimpleTestSuite, TestUCCVerifierSimple,
+        ::testing::Values(
+                UCCVerifierSimpleParams({0}, 1, 12, {{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}},
+                                        "TestFD.csv"),
+                UCCVerifierSimpleParams({0, 1}, 4, 12,
+                                        {{0, 1, 2}, {3, 4, 5}, {6, 7, 8}, {9, 10, 11}},
+                                        "TestFD.csv"),
+                UCCVerifierSimpleParams({0, 1, 2}, 4, 8, {{0, 1}, {3, 4}, {6, 7}, {9, 10}},
+                                        "TestFD.csv"),
+                UCCVerifierSimpleParams({0, 1, 2, 3, 4, 5}, 3, 6, {{3, 4}, {6, 7}, {9, 10}},
+                                        "TestFD.csv"),
+                UCCVerifierSimpleParams({0}, 0, 0, {}, "TestWide.csv"),
+                UCCVerifierSimpleParams({0, 1, 2, 3, 4}, 0, 0, {}, "TestWide.csv"),
+                UCCVerifierSimpleParams({}, 3, 6, {{3, 4}, {6, 7}, {9, 10}}, "TestFD.csv"),
+                UCCVerifierSimpleParams({}, 0, 0, {}, "TestWide.csv")));
+
+TEST_P(TestUCCVerifierWithHyUCC, TestWithHyUCC) {
+    UCCVerifierWithHyUCCParams const& p(GetParam());
+
+    auto hyucc = algos::CreateAndLoadAlgorithm<algos::HyUCC>(p.GetHyUCCParamsMap());
+    hyucc->Execute();
+    std::list<model::UCC> const& mined_uccs = hyucc->UCCList();
+
+    // run ucc_verifier on each UCC from mined_ucc (UCC must hold)
+    for (auto const& current_ucc : mined_uccs) {
+        algos::StdParamsMap ucc_verifier_params_map = p.GetUCCVerifierParamsMap();
+        ucc_verifier_params_map.insert({onam::kUCCIndices, current_ucc.GetColumnIndicesAsVector()});
+        auto verifier = algos::CreateAndLoadAlgorithm<algos::UCCVerifier>(
+                std::move(ucc_verifier_params_map));
+        verifier->Execute();
+        EXPECT_TRUE(verifier->UCCHolds());
+    }
+
+    // Сases of prevent false negative triggering
+    // run ucc_verifier on each UCC from mined_ucc with one column index removed
+    // (UCC must not hold because HyUCC returns minimal UCCs)
```

Comment:
It doesn't check for false negatives, but for false positives, because the code below this comment verifies that `verifier` never returns `true` for non-ucc
```suggestion
    // Сase to prevent false positives
    // run ucc_verifier on each UCC from mined_ucc with one column index removed
    // (UCC must not hold because HyUCC returns minimal UCCs)
```